domain/application/state/unit.go

```diff
@@ -446,6 +446,50 @@ AND a.name = $applicationName.name
 	return result, nil
 }
 
+// GetUnitMachine gets the identifying information for the unit's machine. If the
+// unit does not have a machine assigned, [applicationerrors.UnitMachineNotAssigned] is
+// returned.
+func (st *State) GetUnitMachine(ctx context.Context, unitName coreunit.Name) (application.UnitMachine, error) {
```

Comment:
Split this into two methods. GetUnitMachineUUID and GetUnitMachineName.

Reply:
Imo these methods should be in machine_placement.go as well